recipes/_compute_slurm_finalize.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+#
+# Cookbook Name:: aws-parallelcluster
+# Recipe:: _compute_slurm_finalize
+#
+# Copyright 2013-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+ruby_block 'get_compute_nodename' do
+  block do
+    # Retrieve private hostname from Metadata V2
+    require 'mixlib/shellout'
+    get_hostname_cmd = Mixlib::ShellOut.new('TOKEN=`curl -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 21600"` && curl -H "X-aws-ec2-metadata-token: $TOKEN" -v http://169.254.169.254/latest/meta-data/local-hostname', :user => 'root')
```

Comment:
nit: you could make this more compact by using shellout the same way it's used in fsx_mount:
```
mountname = shell_out!("#{node['cfncluster']['cookbook_virtualenv_path']}/bin/aws fsx --region #{node['cfncluster']['cfn_region']} describe-file-systems --file-system-ids #{node['cfncluster']['cfn_fsx_fs_id']} --query 'FileSystems[0].LustreConfiguration.MountName' --output text", :user=>'root').stdout.strip
```

Reply:
for now you could keep using metadata v1 without the token as done elsewhere

Reply:
I would use the private ip rather than the hostname to retrieve the nodename. it might work better with custom dns setup